ios/AppLovinMAXAdViewManager.m

```diff
@@ -7,297 +7,27 @@
 //
 
 #import "AppLovinMAXAdViewManager.h"
-#import <AppLovinSDK/AppLovinSDK.h>
-#import "AppLovinMAX.h"
-#import <React/RCTUIManager.h>
-
-// Internal
-@interface NSString (ALUtils)
-@property (assign, readonly, getter=al_isValidString) BOOL al_validString;
-@end
-
-@interface AppLovinMAXAdViewManager()
-
-// Dictionaries from id of the React view to the corresponding ad unit id and ad format.
-// Both must be set before the MAAdView is created.
-@property (nonatomic, strong) NSMutableDictionary<NSNumber *, NSString *> *adUnitIdRegistry;
-@property (nonatomic, strong) NSMutableDictionary<NSNumber *, MAAdFormat *> *adFormatRegistry;
-
-// Storage for placement and extra parameters if set before the MAAdView is created
-@property (nonatomic, strong) NSMutableDictionary<NSNumber *, NSString *> *placementRegistry;
-@property (nonatomic, strong) NSMutableDictionary<NSNumber *, NSString *> *customDataRegistry;
-@property (nonatomic, strong) NSMutableDictionary<NSNumber *, NSString *> *adaptiveBannerEnabledRegistry;
-@property (nonatomic, strong) NSMutableDictionary<NSNumber *, NSNumber *> *autoRefreshEnabledRegistry;
-
-@end
+#import "AppLovinMAXAdview.h"
 
 @implementation AppLovinMAXAdViewManager
+
 RCT_EXPORT_MODULE(AppLovinMAXAdView)
 
-// Overridding `init` requires main queue
+RCT_EXPORT_VIEW_PROPERTY(adUnitId, NSString)
+RCT_EXPORT_VIEW_PROPERTY(adFormat, NSString)
+RCT_EXPORT_VIEW_PROPERTY(placement, NSString)
+RCT_EXPORT_VIEW_PROPERTY(customData, NSString)
+RCT_EXPORT_VIEW_PROPERTY(adaptiveBannerEnabled, BOOL)
```

Comment:
These names have to match the Javascript properties so can't be changed since they are already in public.  But, in the view side, we can use isAdaptiveBannerEnabled and isAutoRefreshEnabled as the getters.